python/dgl/graphbolt/impl/csc_sampling_graph.py

```diff
@@ -108,6 +110,58 @@ def total_num_edges(self) -> int:
         """
         return self._c_csc_graph.num_edges()
 
+    @property
+    def num_nodes(self) -> Union[int, Dict[str, int]]:
+        """The number of nodes in the graph.
+        - If the graph is homogenous, returns an integer.
+        - If the graph is heterogenous, returns a dictionary.
+
+        Returns
+        -------
+        Union[int, Dict[str, int]]
+            The number of nodes. Integer indicates the total nodes number of a
+            homogenous graph; dict indicates nodes number per node types of a
+            heterogenous graph.
+
+        Examples
+        --------
+        >>> import dgl.graphbolt as gb, torch
+        >>> total_num_nodes = 5
+        >>> total_num_edges = 12
+        >>> ntypes = {"N0": 0, "N1": 1}
+        >>> etypes = {"N0:R0:N0": 0, "N0:R1:N1": 1,
+        ... "N1:R2:N0": 2, "N1:R3:N1": 3}
+        >>> indptr = torch.LongTensor([0, 3, 5, 7, 9, 12])
+        >>> indices = torch.LongTensor([0, 1, 4, 2, 3, 0, 1, 1, 2, 0, 3, 4])
+        >>> node_type_offset = torch.LongTensor([0, 2, 5])
+        >>> type_per_edge = torch.LongTensor(
+        ... [0, 0, 2, 2, 2, 1, 1, 1, 3, 1, 3, 3])
+        >>> metadata = gb.GraphMetadata(ntypes, etypes)
+        >>> graph = gb.from_csc(indptr, indices, node_type_offset,
+        ... type_per_edge, None, metadata)
+        >>> print(graph.num_nodes)
+        {'N0': tensor(2), 'N1': tensor(3)}
+        """
+
+        offset = self.node_type_offset
+
+        # Homogenous.
+        if offset is None or self.metadata is None:
+            return self._c_csc_graph.num_nodes()
+
+        else:
+            # First, the graph is treated as heterogenous, then the return
+            # type depends on the length of the dictionary.
+            num_nodes_per_type = {
+                _type: offset[_idx + 1] - offset[_idx]
+                for _type, _idx in self.metadata.node_type_to_id.items()
+            }
+
+            if len(num_nodes_per_type) > 1:  # True Heterogenous.
+                return num_nodes_per_type
+            else:  # Homogenous.
+                return self._c_csc_graph.num_nodes()
```

Comment:
It is ok to just return a dict with 1 entry.
```suggestion
            return self._c_csc_graph.num_nodes()
```